samples/archivist_story_runner.py

```diff
@@ -0,0 +1,305 @@
+     - `to_print` is what to print to the console.
+     - `wait_time` is time to wait before running the operation.
+     - `asset_id` is the local reference to the asset, so can be referenced in other operations.
+     - `behaviours` are a list of asset behaviours
+     - `attributes are the asset's attributes
+
+    please see example yaml files for other operation examples.
+
+    To perform all the operations call the `run` method.
+    """
+
+    assets = dict()  # dict of assets created
+    events = list()  # list of events created
+    policies = dict()  # dict of policies created
+
+    def __init__(self, url, token_file, config_file):
+        """
+        constructor
+        :param url: the url of the archivist instance.
+        :param token_file: the token auth for the archivist instance.
+        :param config_file: the yaml config file listing all the operations to perform.
+        """
+
+        with open(token_file, mode="r") as tokenfile:
+            token = tokenfile.read().strip()
+
+            self.client = Archivist(url, auth=token, verify=False)
+
+        # read and save the config file
+        yaml_stream = open(config_file, 'r')
+        self.yaml_object = yaml.load(yaml_stream, Loader=yaml.SafeLoader)
+
+    def run(self):
+        """
+        run runs all operations defined in self.yaml_object
+        """
+        for operation in self.yaml_object["operations"]:
+            wait_time = parse_arg(operation, "wait_time", 0)
+            to_print = parse_arg(operation, "to_print", "")
+
+            self.run_operation(operation["operation"], operation, to_print=to_print,
+                               wait_time=wait_time)
+
+        # cleanup all policies
+        print("Cleanup!")
+        self.delete_all_policies()
+
+    def run_operation(self, operation, operation_args, to_print="", wait_time=0):
+        """
+        run_operation runs an operation based on the given Operation enum.
+        """
+
+        # print out the story of the operation
+        if to_print != "":
+            print(to_print)
+
+        # wait for a number of seconds
+        if wait_time > 0:
+            print("Waiting for {} seconds".format(wait_time))
+            time.sleep(wait_time)
+
+        # find an operation to run
+        if operation == Operation.CREATE_ASSET.name:
+            self.create_asset(operation_args)
+
+        if operation == Operation.CREATE_EVENT.name:
+            self.create_event(operation_args)
+
+        if operation == Operation.CREATE_COMPLIANCE_POLICY.name:
+            self.create_compliance_policy(operation_args)
+
+        if operation == Operation.CHECK_COMPLIANCE.name:
+            self.check_compliance_policy(operation_args)
+
+        if operation == Operation.DELETE_COMPLIANCE.name:
+            self.delete_compliance_policy("policy 1")
+
+        # new line at the end of every operation
+        print("")
+
+    def create_asset(self, *args):
```

Comment:
I dont think this strictly legal in python3 as you have implicitly assumed that args is represented by a dict.
Not good practice - at least asset_id is a required argument.

Reply:
You are 100% correct, I should not have used this from the start.

I have now been explicit in the arg/kwarg list of methods